engine/src/conversion/analysis/struct_accessors.rs

```diff
@@ -0,0 +1,142 @@
+                            continue;
+                        }
+
+                        let body = CppFunctionBody::ReturnFieldAccess(make_ident(field_name));
+
+                        let struct_type = struct_name.name.to_type_path();
+
+                        let fnarg: FnArg = parse_quote! {
+                            this: *const #struct_type
+                        };
+
+                        // TODO: convert arrays to pointers?
+                        let field_type = &field.ty;
+
+                        accessors.push(Api::Function {
+                            name: ApiName::new_from_qualified_name(accessor_name),
+                            fun: Box::new(FuncToConvert {
+                                provenance: Provenance::SynthesizedOther,
+                                ident,
+                                doc_attrs: Vec::new(),
+                                inputs: [fnarg].into_iter().collect(),
+                                variadic: false,
+                                output: parse_quote! {
+                                    -> #field_type
+                                },
+                                vis: parse_quote! { pub },
+                                virtualness: crate::conversion::api::Virtualness::None,
+                                cpp_vis: crate::conversion::api::CppVisibility::Public,
+                                special_member: Some(SpecialMemberKind::GeneratedAccessor),
+                                unused_template_param: false,
+                                references: Default::default(),
+                                original_name: None,
+                                self_ty: None,
+                                synthesized_this_type: None,
+                                add_to_trait: None,
+                                synthetic_cpp: Some((body, CppFunctionKind::Function)),
+                                is_deleted: false,
+                            }),
+                            analysis: (),
+                        })
+                    }
+                }
+                // TODO: is it even possible for these to occur?
+                syn::Fields::Unnamed(_) => todo!(),
+                syn::Fields::Unit => todo!(),
+            };
+
+            // Generate the accessors (if any) + the struct itself
+            Ok(Box::new(accessors.into_iter().chain(std::iter::once(
+                Api::Struct {
+                    name: struct_name,
+                    details,
+                    analysis,
+                },
+            ))))
+        },
+        Api::enum_unchanged,
+        Api::typedef_unchanged,
+    );
+
+    results
+}
+
+fn should_generate_accessor(field: &Field) -> bool {
+    // Don't generate accessors for non-public fields
+    if !matches!(field.vis, Visibility::Public(_)) {
+        return false;
+    }
+
+    // Don't generate accessors for cpp reference fields
+    // TODO: is this restriction necessary?
```

Comment:
I think you're right to err on the side of paranoia for the first iteration here, and bail out in all circumstances where problems could conceivably occur. We can expand coverage later.